### PR TITLE
feat: adding private key argument to patch command

### PR DESCRIPTION
**Pull request by erickzanardo**
## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore
- [ ] 🧪 Tests

### CODE REVIEW

File: packages/shorebird_cli/lib/src/commands/patch/aar_patcher.dart [hunk 1]
```diff
@@ -46,6 +46,9 @@ class AarPatcher extends Patcher {
   @override
   ReleaseType get releaseType => ReleaseType.aar;
 
+  @override
+  Future<void> assertArgsAreValid() async {}
+
```
Comment (bryanoltman) [inline packages/shorebird_cli/lib/src/commands/patch/aar_patcher.dart]: I don't think this is needed

File: packages/shorebird_cli/lib/src/commands/patch/patcher.dart [hunk 2]
```diff
@@ -67,7 +67,7 @@ https://docs.shorebird.dev/status#link-percentage-ios
   Future<void> assertPreconditions();
 
   /// Asserts that the combination arguments passed to the command are valid.
-  Future<void> assertArgsAreValid() async {}
+  Future<void> assertArgsAreValid();
```
Comment (bryanoltman) [inline packages/shorebird_cli/lib/src/commands/patch/patcher.dart]: I'd revert this change. That way, we don't need to provide no-op implementations for the patchers that don't need them.

File: packages/shorebird_cli/lib/src/commands/patch/patch_command.dart [hunk 3]
```diff
@@ -92,6 +92,13 @@ of the iOS app that is using this module.''',
         abbr: 'n',
         negatable: false,
         help: 'Validate but do not upload the patch.',
+      )
+      ..addOption(
+        'private-key-path',
+        hide: true,
+        help: '''
+The path for a private key file that will be used to sign the patch artifact.
+''',
```
Comment (bryanoltman) [inline packages/shorebird_cli/lib/src/commands/patch/patch_command.dart:101]: We should (now or at least before this is announced) add more info about what format this file should be in, how to generate it, etc. Could be a link to docs.

File: packages/shorebird_cli/test/src/commands/patch/aar_patcher_test.dart [hunk 4]
```diff
@@ -115,6 +115,12 @@ void main() {
       patcher = AarPatcher(argResults: argResults, flavor: null, target: null);
     });
 
+    group('assertArgsAreValid', () {
+      test('has no specific validations', () {
+        expect(patcher.assertArgsAreValid, returnsNormally);
+      });
+    });
```
Comment (bryanoltman) [inline packages/shorebird_cli/test/src/commands/patch/aar_patcher_test.dart]: This could be removed too

File: packages/shorebird_cli/test/src/commands/patch/android_patcher_test.dart [hunk 5]
```diff
@@ -436,6 +441,60 @@ Looked in:
...
+            expect(bundle.hashSignature, isNull);
+          }
+        });
+
+        group('when a private key is provided', () {
+          const testPrivateKeyContent = '''
+-----BEGIN PRIVATE KEY-----
+key
+-----END PRIVATE KEY-----
+''';
```
Comment (bryanoltman) [inline packages/shorebird_cli/test/src/commands/patch/android_patcher_test.dart]: I don't think we care about the content of the private key file, just that it exists, right? I think we could get away with writing an empty file, which would make the tests a bit clearer.

File: packages/shorebird_cli/test/src/commands/patch/android_patcher_test.dart [hunk 6]
```diff
@@ -436,6 +441,60 @@ Looked in:
...
+          const mockSignature = 'mock-signature';
```
Comment (bryanoltman) [inline packages/shorebird_cli/test/src/commands/patch/android_patcher_test.dart]: This should be renamed – it's not a `Mock`

File: packages/shorebird_cli/test/src/commands/patch/ios_patcher_test.dart [hunk 7]
```diff
@@ -138,6 +138,12 @@ void main() {
         );
       });
 
+      group('assertArgsAreValid', () {
+        test('has no specific validations', () {
+          expect(patcher.assertArgsAreValid, returnsNormally);
+        });
+      });
+
       group('archiveDiffer', () {
```
Comment (bryanoltman) [inline packages/shorebird_cli/test/src/commands/patch/ios_patcher_test.dart:147]: This can be removed

File: packages/shorebird_cli/test/src/commands/patch/patcher_test.dart [hunk 8]
```diff
@@ -72,4 +72,9 @@ class _TestPatcher extends Patcher {
 
   @override
   ReleaseType get releaseType => throw UnimplementedError();
+
+  @override
+  Future<void> assertArgsAreValid() {
+    throw UnimplementedError();
+  }
 }
```
Comment (bryanoltman) [inline packages/shorebird_cli/test/src/commands/patch/patcher_test.dart]: This can be removed

File: packages/shorebird_cli/test/src/commands/patch/android_patcher_test.dart [hunk 9]
```diff
@@ -436,6 +441,60 @@ Looked in:
...
+            );
+
+            const expectedSignatures = [
+              mockSignature,
+              mockSignature,
+              mockSignature,
+            ];
+
+            final signatures =
+                result.values.map((bundle) => bundle.hashSignature).toList();
```
Comment (bryanoltman) [inline packages/shorebird_cli/test/src/commands/patch/android_patcher_test.dart]: Might be useful to verify that we're generating a signature per hash? This reads like we might be using the same signature for every hash, which would (in most cases) not be right.

File: packages/shorebird_cli/lib/src/commands/patch/patcher.dart (hunk 2 above)
Comment (erickzanardo) [reply packages/shorebird_cli/lib/src/commands/patch/patcher.dart]: Yeah, but I made that change because otherwise coverage would complain about the default (empty) implementation not being covered. Happy to revert it though if there is a better approach

**Review by bryanoltman — APPROVED** [review]
